doc/how.rst

```diff
@@ -3,87 +3,160 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. literalinclude:: ../benchopt/tests/test_benchmarks/dummy_benchmark/datasets/boston.py
 
-You can also define a parametrized dataset for example to test
+You can also define a parametrized simulated dataset for example to test
 across problem dimensions.
 
-Example of parametrized dataset
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Example of parametrized simulated dataset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Using simulated data:
+Sometimes one wants to test the solvers for variants of the same dataset.
+For example, one may want to change the dataset size, the noise level etc.
+To be able to specify parameters to get a dataset you can use a class
+attribute called `parameters`. This parameter must be a dictionary
+whose keys can be passed to the `__init__` of the class. The benchopt
+will automatically allow you to test all combinations of parameters.
 
 .. literalinclude:: ../benchopt/tests/test_benchmarks/dummy_benchmark/datasets/simulated.py
 
+.. _solvers:
+
 3. Solvers
 ----------
 
-Solver requires to define a `set_objective` that is constructed
-from the union of the data and the objective parameters.
+Solver requires to define three methods:
+
+   - **set_objective**: This method expects as input the union of what is returned
+     by the **get_data** methods of both the dataset and the objective.
 
-You can define your custom solver.
+   - **run**: This method takes only one parameter that controls the stopping
+     condition of the solver. This is typically a number of iterations
+     or a tolerance parameter.
 
-Example of hand written and parametrized solver
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   - **get_result**: This method returns a variable that can be passed
+     to the **compute** method from the objective. This is the value of
+     the iterates.
 
-Using a hand written solver:
+benchopt supports different types of solvers:
+
+   - :ref:`python_numpy_solvers`
+   - :ref:`python_conda_solvers`
+   - :ref:`python_source_solvers`
+   - :ref:`r_solvers`
+   - :ref:`julia_solvers`
+
+.. _python_numpy_solvers:
+
+Python solver based on Numpy, Scipy, Numba
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Such solvers are written in pure `Python <https://www.python.org/>`_ without any compiled
+code. They are typically written in `Numpy <https://numpy.org/>`_ and possibly
+with some just in time compilation e.g. with `Numba <https://numba.pydata.org/>`_.
 
 .. literalinclude:: ../benchopt/tests/test_benchmarks/dummy_benchmark/solvers/python_pgd.py
 
-Example of a solver available from a package on conda
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _python_conda_solvers:
+
+Python solver from Conda package
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Using a package available from conda:
+If you want to compare a solver available on conda you can specify
+how this solver needs to be installed and how to call it.
+The `install_cmd` class variable needs to be set to `conda`
```

Comment:
`"conda"` ?

Reply:
Here, this also work with `pip` installable solvers.